### PR TITLE
Feat/LIVE-7596: Genuine check and firmware update check in Early Security Check during Sync Onboarding

### PR DESCRIPTION
**Pull request by alexandremgo**
### 📝 Description

[LIVE-7596](https://ledgerhq.atlassian.net/browse/LIVE-7596): During the "Early Security Check" step of the onboarding of a Stax, before the synchronous onboarding companion step:
- added genuine check
- added firmware update check

[LIVE-7597](https://ledgerhq.atlassian.net/browse/LIVE-7597): UI of the ESC
- display information and error messages during the genuine check and the 
- interactions with the user in drawers 
- Note: genuine check failing for other reasons than the user refused on the Stax is not correctly handled 

[LIVE-7600](https://ledgerhq.atlassian.net/browse/LIVE-7600): When a fw update is available, the user is able to go to the firmware update flow
- if successful update, or if they stops the fw update: it goes back to the ESC
- once coming back to the ESC, the genuine check is already marked as done, and a check on the available firmware update is triggered

[LIVE-8360](https://ledgerhq.atlassian.net/browse/LIVE-8360): adapt the fw update flow to the ESC:
- removed restoring apps (in case from ESC)
- removed drawer asking the user to wait and enter their pin code (in case from ESC)
- KEEP the success screen at the end, with a button to return to the ESC

[LIVE-8846](https://ledgerhq.atlassian.net/browse/LIVE-8846): handle errors (and non genuine device) during the genuine check

Other changes:
- `onBackFromUpdate` prop for the screen `FirmwareUpdate` (and `FirmwareUpdateBanner`) now takes a `updateState: UpdateStep` param so we can know if the firmware update was completed before leaving the screen or if the user left the firmware update uncompleted
- on polling hook `useOnboardingStatePolling`: returns a function to reset the polling state

### ❓ Context

- **Impacted projects**: `LLM`
- **Linked resource(s)**: [LIVE-7596](https://ledgerhq.atlassian.net/browse/LIVE-7596) & [LIVE-7597](https://ledgerhq.atlassian.net/browse/LIVE-7597) & [LIVE-7600](https://ledgerhq.atlassian.net/browse/LIVE-7600) & [LIVE-8360](https://ledgerhq.atlassian.net/browse/LIVE-8360) & [LIVE-8846](https://ledgerhq.atlassian.net/browse/LIVE-8846)

### ✅ Checklist

- [ ] **Test coverage** To be QAed
- [x] **Atomic delivery** 
- [ ] **No breaking changes** 

### 📸 Demo

<details>
  <summary>Small demo nominal path on Android</summary>

https://github.com/LedgerHQ/ledger-live/assets/15096913/6388449e-dbec-4e44-bbee-40201921d2f8

</details>
### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

[LIVE-7596]: https://ledgerhq.atlassian.net/browse/LIVE-7596?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LIVE-7597]: https://ledgerhq.atlassian.net/browse/LIVE-7597?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LIVE-7600]: https://ledgerhq.atlassian.net/browse/LIVE-7600?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LIVE-8360]: https://ledgerhq.atlassian.net/browse/LIVE-8360?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-mobile/src/screens/SyncOnboarding/AllowManagerDrawer.tsx [hunk 1]
```diff
@@ -1,18 +1,34 @@
...
+
+  /**
+   * Callback when the drawer is closed
+   *
+   * As the drawer has no close button (`noCloseButton`), the drawer is closed only when `isOpen === false`
+   */
   onClose?: () => void;
+
+  /**
+   * A `Device` object
```
Comment (ofreyssinet-ledger) [inline apps/ledger-live-mobile/src/screens/SyncOnboarding/AllowManagerDrawer.tsx:24]: thank you 😆

File: apps/ledger-live-mobile/src/screens/SyncOnboarding/AllowManagerDrawer.tsx [hunk 2]
```diff
@@ -25,13 +41,13 @@ const AllowManagerDrawer = ({ isOpen, device, onClose }: Props) => {
...
     >
-      <Flex mb={260} pt={110}>
-        <AllowManager
-          t={t}
-          wording={t("syncOnboarding.softwareChecksSteps.allowManagerDrawer.wording")}
-          device={device}
-          theme={theme}
-        />
+      <Flex mt={8} alignItems="center" justifyContent="center">
+        <Text fontWeight="semiBold" fontSize={24} textAlign="center">
```
Comment (ofreyssinet-ledger) [inline apps/ledger-live-mobile/src/screens/SyncOnboarding/AllowManagerDrawer.tsx:45]: you're not supposed to specify fontSize, use a `variant` and if there is no `variant` specified in the design you should gently ask the designer to add one

File: apps/ledger-live-mobile/src/screens/SyncOnboarding/DesyncOverlay.tsx [hunk 3]
```diff
@@ -66,7 +66,7 @@ export const PlainOverlay = ({ isOpen, delay }: DelayProps) => (
   </DisplayWithDelay>
 );
 
-const ResyncOverlay = ({ isOpen, delay = 0, productName }: Props) => {
+const DesyncOverlay = ({ isOpen, delay = 0, productName }: Props) => {
```
Comment (ofreyssinet-ledger) [inline apps/ledger-live-mobile/src/screens/SyncOnboarding/DesyncOverlay.tsx:69]: maybe a `/** description of the component */` would be nice here

File: apps/ledger-live-mobile/src/screens/SyncOnboarding/index.tsx [hunk 4]
```diff
@@ -185,14 +249,20 @@ export const SyncOnboarding = ({ navigation, route }: SyncOnboardingScreenProps)
...
   const handleDesyncClose = useCallback(() => {
-    setDesyncDrawerOpen(false);
+    setIsDesyncDrawerOpen(false);
     navigation.goBack();
   }, [navigation]);
 
+  const handleAutoRepairClose = useCallback(() => {
+    setIsAutoRepairOpen(false);
+    setIsPollingOn(true);
+    // TODO: reset polling ?
```
Comment (ofreyssinet-ledger) [inline apps/ledger-live-mobile/src/screens/SyncOnboarding/index.tsx:263]: leftover TODO ?

File: apps/ledger-live-mobile/src/screens/SyncOnboarding/index.tsx [hunk 5]
```diff
@@ -129,20 +175,28 @@ export const SyncOnboarding = ({ navigation, route }: SyncOnboardingScreenProps)
...
   }, [fatalError]);
 
   // An allowed error during polling (which makes the polling retry) only triggers an error message after a timeout
   useEffect(() => {
     let timeout: ReturnType<typeof setTimeout>;
 
-    if (allowedError) {
-      timeout = setTimeout(() => {
+    if (allowedError && !(allowedError instanceof LockedDeviceError)) {
+      // TODO: not a DeviceExtractOnboardingStateError but a specific error.
```
Comment (ofreyssinet-ledger) [inline apps/ledger-live-mobile/src/screens/SyncOnboarding/index.tsx:189]: leftover TODO ?

File: apps/ledger-live-mobile/src/screens/SyncOnboarding/EarlySecurityCheck.tsx [hunk 6]
```diff
@@ -1,42 +1,657 @@
...
+type CheckCardProps = FlexBoxProps & {
+  title: string;
+  description?: string | null;
+  learnMore?: string | null;
+  onLearnMore?: () => void;
+  index: number;
+  status: UiCheckStatus;
+};
+
+const CheckCard = ({
```
Comment (ofreyssinet-ledger) [inline apps/ledger-live-mobile/src/screens/SyncOnboarding/EarlySecurityCheck.tsx:597]: possible to move this in a separate file?

File: apps/ledger-live-mobile/src/screens/SyncOnboarding/EarlySecurityCheck.tsx [hunk 7]
```diff
@@ -1,42 +1,657 @@
...
+   * Function to cancel the onboarding
+   */
+  onCancelOnboarding: () => void;
 };
 
 /**
  * Component representing the early security checks step, which polls the current device state
  * to display correctly information about the onboarding to the user
  */
 export const EarlySecurityCheck: React.FC<EarlySecurityCheckProps> = ({
```
Comment (ofreyssinet-ledger) [inline apps/ledger-live-mobile/src/screens/SyncOnboarding/EarlySecurityCheck.tsx:91]: can we split this huge component in a `ComponentA` handling the logic and a `ComponentB` with 0 logic and just rendering stuff?
similar to here https://github.com/LedgerHQ/ledger-live/pull/3909/files#diff-dff85510b1a6ee2a82f5094b9dc0cb7298e1dd8c1a2a3f3ff62bee26e1bf16ab

File: apps/ledger-live-mobile/src/screens/SyncOnboarding/index.tsx (hunk 4 above)
Comment (alexandremgo) [reply apps/ledger-live-mobile/src/screens/SyncOnboarding/index.tsx:263]: Only forgot to remove the comment on this one

File: apps/ledger-live-mobile/src/screens/SyncOnboarding/index.tsx (hunk 5 above)
Comment (alexandremgo) [reply apps/ledger-live-mobile/src/screens/SyncOnboarding/index.tsx:189]: A comment i forgot to remove, sorry

File: apps/ledger-live-mobile/src/screens/SyncOnboarding/EarlySecurityCheck.tsx (hunk 7 above)
Comment (alexandremgo) [reply apps/ledger-live-mobile/src/screens/SyncOnboarding/EarlySecurityCheck.tsx:84]: Done ;)

File: apps/ledger-live-mobile/src/screens/SyncOnboarding/AllowManagerDrawer.tsx [hunk 8]
```diff
@@ -42,7 +42,7 @@ const AllowManagerDrawer = ({ isOpen, device, onClose }: Props) => {
       noCloseButton
     >
       <Flex mt={8} alignItems="center" justifyContent="center">
-        <Text fontWeight="semiBold" fontSize={24} textAlign="center">
+        <Text variant="h4" textAlign="center">
```
Comment (ofreyssinet-ledger) [inline apps/ledger-live-mobile/src/screens/SyncOnboarding/AllowManagerDrawer.tsx:45]: didn't test but double check if you shouldn't use `h4Inter` instead (this one might be the big ass capitalized font)

File: apps/ledger-live-mobile/src/screens/SyncOnboarding/EarlySecurityCheck.tsx [hunk 9]
```diff
@@ -1,42 +1,405 @@
...
+      }
+
+      // Updates the UI
+      if (getLatestAvailableFirmwareStatus === "available-firmware" && latestFirmware) {
+        // Only for QA to have always the same 1-way path: on infinite loop firmware
+        // the path 1.2.1-il2 -> 1.2.1-il0 does not trigger a firmware update during the ESC
+        if (
+          getLatestAvailableFirmwareStatus === "available-firmware" &&
+          latestFirmware?.final.name === "1.2.1-il0" &&
+          deviceInfo?.version === "1.2.1-il2"
```
Comment (ofreyssinet-ledger) [inline apps/ledger-live-mobile/src/screens/SyncOnboarding/EarlySecurityCheck.tsx:331]: maybe make it work like that for all possible versions of IL fw?

Comment (alexandremgo) [reply apps/ledger-live-mobile/src/screens/SyncOnboarding/EarlySecurityCheck.tsx:330]: perfectionniste 🫣 done ;)

File: apps/ledger-live-mobile/src/screens/SyncOnboarding/AllowManagerDrawer.tsx (hunk 8 above)
Comment (alexandremgo) [reply apps/ledger-live-mobile/src/screens/SyncOnboarding/AllowManagerDrawer.tsx:45]: I don't think i can use `h4Inter` variant, only `h1Inter` seems to exist. But anyway `h4` is not a big ass capitalized font (`h3` is though)

File: apps/ledger-live-mobile/src/screens/SyncOnboarding/EarlySecurityCheck.tsx [hunk 10]
```diff
@@ -1,42 +1,404 @@
...
+          getLatestAvailableFirmwareError.type === "SharedError" &&
+          getLatestAvailableFirmwareError.retrying
+        )
+      ) {
+        log(
+          "EarlySecurityCheck",
+          "Failed to retrieve latest firmware version with error:",
+          getLatestAvailableFirmwareError.name,
+        );
+        setFirmwareUpdateCheckStatus("error");
```
Comment (ofreyssinet-ledger) [inline apps/ledger-live-mobile/src/screens/SyncOnboarding/EarlySecurityCheck.tsx:307]: I thought this was not really allowed in react (updating state inside the render function) and thought you should instead use a `useEffect` but it seems like it's actually ok in some cases like that https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes
Makes me think I know nothing about good practices in React 🤔🙈

File: apps/ledger-live-mobile/src/components/ChangeDeviceLanguageAction.tsx [hunk 11]
```diff
@@ -15,6 +15,13 @@ type Props = {
   onStart?: () => void;
   onResult?: () => void;
   onError?: (err: Error) => void;
+  /*
+   * Defines in what type of component this action will be rendered.
+   *
+   * If "drawer", the component will be rendered as a content to be rendered in a drawer.
+   * If "view", the component will be rendered as a view. Defaults to "view".
+   */
+  renderedInType?: "drawer" | "view";
```
Comment (ofreyssinet-ledger) [inline apps/ledger-live-mobile/src/components/ChangeDeviceLanguageAction.tsx:24]: I feel like if this is only needed by the `BluetoothDisabled` component we could avoid to add this prop that has to be passed down everywhere (prop drilling) and instead use a simple context:
- `RenderedInTypeContext = React.createContext({renderedInType: "view"})`
- default value of the context is "view"
- wrap content of drawers (all except one in LLM are implemented with `QueuedDrawer` I think so just 1 place to change) with <RenderedInTypeContext.Provider value={{renderedInType: "drawer"}) />
- use in `BluetoothDisabled`: `const {renderedInType} = useContext(RenderedInTypeContext)`

Comment (ofreyssinet-ledger) [reply apps/ledger-live-mobile/src/components/ChangeDeviceLanguageAction.tsx:24]: also maybe instead of "view" use "fullScreen" or something less generic

Comment (ofreyssinet-ledger) [reply apps/ledger-live-mobile/src/components/ChangeDeviceLanguageAction.tsx:24]: or maybe better with just a boolean like `isInDrawer: boolean` 🤷‍♂️

File: apps/ledger-live-mobile/src/screens/SyncOnboarding/AllowManagerDrawer.tsx (hunk 8 above)
Comment (ofreyssinet-ledger) [reply apps/ledger-live-mobile/src/screens/SyncOnboarding/AllowManagerDrawer.tsx:45]: ah my bad I'm dumb I was checking react-ui 😆

**Review by ofreyssinet-ledger — APPROVED** [review]
small comments but nothing blocking I think, lgtm 👍

File: apps/ledger-live-mobile/src/components/ChangeDeviceLanguageAction.tsx (hunk 11 above)
Comment (alexandremgo) [reply apps/ledger-live-mobile/src/components/ChangeDeviceLanguageAction.tsx:24]: That could be nice to use a context yeah

File: apps/ledger-live-mobile/src/screens/SyncOnboarding/EarlySecurityCheck.tsx (hunk 10 above)
Comment (alexandremgo) [reply apps/ledger-live-mobile/src/screens/SyncOnboarding/EarlySecurityCheck.tsx:307]: Yeah it's actually from this that i got the idea

**Pull request merged**